Source/quests.cpp

```diff
@@ -113,15 +113,12 @@ int QuestGroup4[2] = { Q_VEIL, Q_WARLORD };
 
 void InitQuests()
 {
-	int i, initiatedQuests;
-	DWORD z;
-
 	if (!gbIsMultiplayer) {
-		for (i = 0; i < MAXQUESTS; i++) {
-			quests[i]._qactive = QUEST_NOTAVAIL;
+		for (auto &quest : quests) {
+			quest._qactive = QUEST_NOTAVAIL;
 		}
 	} else {
-		for (i = 0; i < MAXQUESTS; i++) {
+		for (int i = 0; i < MAXQUESTS; i++) {

```

Comment:
This is done automatically where applicable. The reason it happens on 117 is that once `i` is limited to the loop, clang-tidy can determine that it will is only used to iterate over the full content of `quests`. The reason that it does not happen to 121 is that `i` is used as an index in both `questlist` and `quests` and so the loop can not be simply converted to a range.

Reply:
Gotcha. Oh well...

Reply:
Yeah I always get disappointed with situations like this one, so close but not quite there.